### PR TITLE
HTCONDOR-2140 no-simplelists

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by GregThain** [comment]
Note this is failing on some linux platforms in batlab with

```
[ 16%] Built target test_htcondor_annex_create_constraints
/var/lib/condor/execute/slot1/dir_3462850/userdir/build-Z9Unmx/BUILD/condor-23.3.0/src/condor_daemon_client/daemon_list.cpp: In member function 'int CollectorList::resortLocal(const char*)':
/var/lib/condor/execute/slot1/dir_3462850/userdir/build-Z9Unmx/BUILD/condor-23.3.0/src/condor_daemon_client/daemon_list.cpp:103:14: error: 'sort' is not a member of 'std'; did you mean 'sqrt'?
  103 |         std::sort(m_list.begin(), m_list.end(),
      |              ^~~~
      |              sqrt
```

I've seen this before, we're missing an #include <algorithm>, and I'm frequently bitten by it.  I don't know if there's some unix'es where we leak a #include <algorithm> indirectly from condor_common.h, or if one of the system includes leaks it. 

If you want to be a hero, you could figure out where the include leak happens and fix it, so we don't keep running into this problem.  Or just add the missing include.

**Review by GregThain — CHANGES_REQUESTED** [review]

File: src/condor_schedd.V6/schedd.cpp [hunk 1]
```diff
@@ -13393,20 +13357,22 @@ Scheduler::Init()
...
-		FlockCollectors = new DaemonList();
-		FlockCollectors->init( DT_COLLECTOR, flock_collector_hosts );
-		MaxFlockLevel = FlockCollectors->number();
+		MaxFlockLevel = FlockCollectors.size();
 		MinFlockLevel = param_integer("MIN_FLOCK_LEVEL", 0, 0, MaxFlockLevel);
 
-		if( FlockNegotiators ) {
-			delete FlockNegotiators;
+		FlockNegotiators.clear();
+		size_t idx = 0;
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_schedd.V6/schedd.cpp:13368]: Oops. Fixed pushed.

File: src/condor_schedd.V6/schedd.cpp [hunk 2]
```diff
@@ -14502,13 +14463,10 @@ Scheduler::sendReschedule( int /* timerID */ )
...
 	negotiator->sendMsg( msg.get() );
 
-	Daemon* d;
-	if( FlockNegotiators ) {
-		FlockNegotiators->rewind();
-		FlockNegotiators->next( d );
-		for( int i=0; d && i<FlockLevel; FlockNegotiators->next(d), i++ ) {
-			st = d->hasUDPCommandPort() ? Stream::safe_sock : Stream::reli_sock;
-			negotiator = new Daemon( *d );
+	if( FlockNegotiators.size() ) {
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_schedd.V6/schedd.cpp:14467]: Interesting. I'll have to look into that.

**Review by GregThain — APPROVED** [review]
LGTM.

**Pull request merged**